nautilus_trader/common/component.pyx

```diff
@@ -2596,7 +2596,17 @@ cdef class MessageBus:
         return subs_array
 
 
+cdef inline bint contains_wildcard(str topic_or_pattern):
+    return '?' in topic_or_pattern or '*' in topic_or_pattern
+
+
 cdef inline bint is_matching(str topic, str pattern):
+    topic_contains_wildcard = contains_wildcard(topic)
```

Comment:
Small Cython thing: if a local variable isn't typed with a `cdef` then the Python interpreter is needed to figure out the type. Will merge like this though and optimize on `develop`.